### PR TITLE
air-quality-monitor-app

### PR DESCRIPTION
**Pull request by GedeonTS**
# air-quality-monitor-app
In this pull I created a single page app(SPA) that displays data from the air pollution API and created tests.

### working interactions

- [ ]  Home page

- When the page loads, the web app shows the list of city's
- Along with the city name are the geographical coordinates that come from the API.
- When the user clicks (or taps) on a city item, the application navigates to the details page.

- [ ] Details page

- In the details page, the webapp retrieves data from the API to show detailed data in the city.
When the user clicks on the "Back" button (<), the user navigates to the home page.

Here is the [descrition video](https://www.loom.com/share/367a5e34e3f94e51abc5c0eb27c2ee54)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,14 +1,42 @@
...
+    <Routes>
+      <Route path="/paris" element={<Paris />} />
+      <Route path="/Warsaw" element={<Warsaw />} />
+      <Route path="/Barcelona" element={<Barcelona />} />
+      <Route path="/Rome" element={<Rome />} />
+      <Route path="/Lisbon" element={<Lisbon />} />
+      <Route path="/Madrid" element={<Madrid />} />
+      <Route path="/London" element={<London />} />
+      <Route path="/Brussels" element={<Brussels />} />
+      <Route path="/" element={<Home coord={coord} />} />
```
Comment (andersonlebon) [inline src/App.js:37]: ## [OPTIONAL]
- [x] Nice job :clap: on navigating between pages without reloading. However, I have seen all of your components(Barcelona, London, Madrid,...) have the same structure. The better approach I can suggest is to use reusable components as this is one of the beauties of `Reactjs`.

File: src/components/redux/configStore.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import coordReducer from './coord';
+
+const allReducers = combineReducers({
+  data: dataReducer,
+  coord: coordReducer,
+});
+
+const store = createStore(allReducers, applyMiddleware(thunk));
+
+export default store;
```
Comment (andersonlebon) [inline src/components/redux/configStore.js]: ## [OPTIONAL]

- [x] To structure well your project I would really suggest you move the `/redux` directory outside the `/components` directory, as redux only cares about the store and not components. This structure example would help :point_down: 

```
app-root-directory
  |- package.json
  |- README.md
  |- /public
    |- index.html
  |- /src
    |- app.css
    |- index.js
    |- /components
    |- /container
    |- /pages
    |- /redux
  |- /node_modules
```

File: src/__tests__/actions.test.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+it('should execute fetch coords', () => {
+  const store = mockStore({});
+
+  return store.dispatch(fetchCoord())
+    .then(() => {
+      const actions = store.getActions();
+      expect(actions[0]).toEqual(getCoord());
+    });
+});
```
Comment (andersonlebon) [inline src/__tests__/actions.test.js]: - [x] Kindly note that some of your tests are failing. Would you please fix them?

image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @GedeonTS 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/redux/configStore.js (hunk 2 above)
Comment (GedeonTS) [reply src/components/redux/configStore.js]: okay

File: src/__tests__/actions.test.js (hunk 3 above)
Comment (GedeonTS) [reply src/__tests__/actions.test.js]: Hey @andersonlebon, on my side all tests are running well, is there any solution you can suggest me?

File: src/App.js (hunk 1 above)
Comment (GedeonTS) [reply src/App.js:37]: cool

File: src/__tests__/city.test.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import city from '../__mocks__/city';
+
+describe('display missions component', () => {
+  test('snapshot renders properly', () => {
+    expect(city).toMatchSnapshot();
+  });
+});
```
Comment (julie-ify) [inline src/__tests__/city.test.js:7]: - [ ] All your tests are well written and passed.
chrome_EAOxtYAQuP

**Review by julie-ify — APPROVED** [review]
Hi @GedeonTS, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**